include/seqan3/range/container/constexpr_string.hpp

```diff
@@ -0,0 +1,595 @@
+    {
+        return !(*this == rhs);
+    }
+
+    /*!\brief Compares two strings lexicographically.
+     * \param rhs string whose content is compared to `this`.
+     * \returns `bool` `true` if the corresponding comparison holds, `false` otherwise.
+     *
+     * ### Complexity
+     *
+     * Linear in the size of the strings.
+     */
+    template <size_t N2>
+    constexpr bool operator<(constexpr_string<N2> const & rhs) const noexcept
+    {
+        for (unsigned i = 0; i < ((N < N2) ? N : N2); ++i)
+        {
+            if (_lit[i] < rhs._lit[i])
+                return true;
+            else if (_lit[i] != rhs._lit[i])
+                return false;
+        }
+        return N < N2;
+    }
+
+    /*!\brief Compares two strings lexicographically.
+     * \param rhs string whose content is compared to `this`.
+     * \returns `bool` `true` if the corresponding comparison holds, `false` otherwise.
+     *
+     * ### Complexity
+     *
+     * Linear in the size of the strings.
+     */
+    template <size_t N2>
+    constexpr bool operator<=(constexpr_string<N2> const & rhs) const noexcept
+    {
+        for (unsigned i = 0; i < ((N < N2) ? N : N2); ++i)
+        {
+            if (_lit[i] > rhs._lit[i])
+                return false;
+        }
+        return N <= N2;
+    }
+
+    /*!\brief Compares two strings lexicographically.
+     * \param rhs string whose content is compared to `this`.
+     * \returns `bool` `true` if the corresponding comparison holds, `false` otherwise.
+     *
+     * ### Complexity
+     *
+     * Linear in the size of the strings.
+     */
+    template <size_t N2>
+    constexpr bool operator>(constexpr_string<N2> const & rhs) const noexcept
+    {
+        return !(*this <= rhs);
+    }
+
+    /*!\brief Compares two strings lexicographically.
+     * \param rhs string whose content is compared to `this`.
+     * \returns `bool` `true` if the corresponding comparison holds, `false` otherwise.
+     *
+     * ### Complexity
+     *
+     * Linear in the size of the strings.
+     */
+    template <size_t N2>
+    constexpr bool operator>=(constexpr_string<N2> const & rhs) const noexcept
+    {
+        return !(*this < rhs);
+    }
+
+    /*!\brief Swaps the contents.
+     *
+     * \details
+     *
+     * Both strings must have the same size in order to swap them.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     */
+    constexpr void swap(constexpr_string & other) noexcept
+    {
+        std::swap(*this, other);
+    }
+    //!\}
+};
+
+/*!\brief Exchanges the given values.
+ * \relates constexpr_string
+ * \tparam N The size of the constexpr_string.
+ * \param lhs The left operand.
+ * \param rhs The right operand.
+ *
+ * ### Exceptions
+ *
+ * No-throw guarantee.
+ *
+ * ### Complexity
+ *
+ * Constant.
+ */
+template <size_t N>
+constexpr inline void
+swap(constexpr_string<N> & lhs, constexpr_string<N> & rhs)
+{
+    lhs.swap(rhs);
+}
+
+/*!\name Deduction guides
```

Comment:
Does this grouping actually work? I had problems with them outside of class scope...